### PR TITLE
update line-through for asciidoc writer to custom inline style

### PR DESCRIPTION
**Pull request by kbroch-rivosinc**
issue:  #8933

### CODE REVIEW

**Comment by jgm** [comment]
Q: Is this an asciidoctor-only feature?
If it only works with asciidoctor and not asciidoc, we need to put this under a conditional.

**Comment by kbroch-rivosinc** [comment]
> Q: Is this an asciidoctor-only feature? If it only works with asciidoctor and not asciidoc, we need to put this under a conditional.

I thiink it is an asciidoc feature, but here's detailed pros/cons:

* pro: in asciidoc spec: https://docs.asciidoctor.org/asciidoc/latest/text/custom-inline-styles/
* con: it is css controlled at end of the day
* pro: works with asciidoctor-pdf out of the box: `asciidoctor-pdf strke-through.adoc`
[strke-through.pdf](https://github.com/jgm/pandoc/files/11924982/strke-through.pdf)

Wondering of @mojavelinux could add any more insights?

**Comment by mojavelinux** [comment]
line-through has always been recognized as a built-in role in AsciiDoc. The default stylesheet/theme will apply the line-through text decoration to this span of text.

(In the planned semantic HTML converter, we will likely be switching to the `<s>` tag in the HTML output. However, that's a presentation concern and thus won't change the AsciiDoc source).

**Comment by mojavelinux** [comment]
> pro: in asciidoc spec: docs.asciidoctor.org/asciidoc/latest/text/custom-inline-styles

FYI: This is not the AsciiDoc spec. This is the initial contribution to the AsciiDoc spec based on the defacto implementation of AsciiDoc (how it is implemented in Asciidoctor). Once the first version of the AsciiDoc spec is completed, the documentation at that URL will become the user-facing guide for AsciiDoc (and may be a mirror of the official guide).

Please note that there is no "asciidoc vs asciidoctor". There is only one AsciiDoc. It's currently in the process of being formalized as a specification. Asciidoctor is the reference for that specification and will become an implementation once there is a specification to implement. It's possible there will be changes during this time, but whatever is decided on becomes AsciiDoc and there is nothing else.

**Comment by jgm** [comment]
When I say `asciidoc` I'm referring to the older Python implementation, which last I checked was still being used (e.g. by git for its documentation), and which has somewhat different syntax in places. Pandoc distinguishes between `asciidoc` and `asciidoctor` because of this.

If this is no longer the case, and the old implementation is not being used by anyone, we could adjust this and unify the syntax so that `asciidoc` = `asciidoctor`.

Remember that one important use of pandoc is converting legacy documentation, so even if the old python `asciidoc` isn't much used any more, it's valuable to be able to support conversion from old documentation bases that used it.

**Comment by kbroch-rivosinc** [comment]
@jgm IMO there's not a lot of value in supporting a doc conversion workflow that still wants to use the old asciidoc (py) format as output (there isn't any support for asciidoc(tor) in input, right?): 

* New doc workflows would choose, to the new supported format. 
*  If for some reason I had a workflow that converted to old asciidoc format in the middle to support some other tooling, I'd be worried this would be deprecated and eventually EOL'ed.

**Comment by mojavelinux** [comment]
I interact with the AsciiDoc community on a daily basis, and I'm not aware of anyone who's still actively using AsciiDoc.py to convert content being written today. Naturally, older documents still exist that were written during the time of AsciiDoc.py and, as a result, may still rely on it to be converted. However, no one should be generating new documents with AsciiDoc.py as a target. This is even acknowledged on that project's website: "AsciiDoc.py is a legacy processor for this syntax".

As the creator and maintainer of Asciidoctor, I would prefer not to see an asciidoctor option in pandoc. As the lead of the AsciiDoc specification, I'm especially keen on not letting the impression to take hold that there are flavors of AsciiDoc. In fact, it's written into the charter of our group to prevent that from happening.

For trademark reasons, the AsciiDoc.py project has be renamed to asciidoc-py. What that means is that AsciiDoc.py is no longer AsciiDoc. To reiterate, there's only one AsciiDoc, and it's actively being formalized as a specification based on the syntax supported by Asciidoctor. For the past decade leading up to current specification effort, Asciidoctor has been the maintainer of the definition of AsciiDoc.

If you feel strongly that it's important to retain an output format that targets AsciiDoc.py (which is understandable), then I would encourage you to rename it to asciidoc-py. That would mean the output format named asciidoctor should become renamed to (or aliased to) asciidoc. Given this is not my project, I acknowledge the the choice is not mine to make. But I would hope you would take my recommendation under consideration.

If you have questions or concerns about what I've stated here, I'm glad to provide clarifications. You're also invited to ask those question in the project chat for AsciiDoc at https://chat.asciidoc.org.

**Comment by jgm** [comment]
> As the creator and maintainer of Asciidoctor, I would prefer not to see an asciidoctor option in pandoc. 

There already is one.  That's because pandoc started supporting asciidoc before asciidoctor existed. When asciidoctor came along with some incompatible syntax, we differentiated by adding `asciidoctor` as a separate writer name.

We could rename things so that `asciidoc` does what `asciidoctor` used to and `asciidoc_legacy` or `asciidoc_py` does what `asciidoc` used to.  I will think about this, but of course there is a cost: possibly breaking workflows that currently work.  In any event, there should probably be a separate issue to track this; I'll set one up.

**Comment by mojavelinux** [comment]
Thanks John.

**Comment by jgm** [comment]
Now that we've gotten clear about terminology, let me rephrase my original question.  Is the output for strikethrough produced by this PR suitable for *both* new asciidoc (asciidoctor) and legacy python asciidoc?

If not, then we need some conditionals so we can give good results for both.

You can determine which variant you're using by doing
```hs
  isAsciidoctor <- gets asciidoctorVariant                         
```
Of course, if we implement the renaming discussed above, these internal names may also change, but for purposes of this PR let's go with what we have.

**Comment by mojavelinux** [comment]
For the purpose of this issue, there is no difference in any event. The `line-through` role for strikethrough has been supported in the AsciiDoc syntax since it's inception.

**Comment by jgm** [comment]
The build failures are not related to this PR. I think rebasing against main should give you a clean build and then we can merge.
